pipelinerl/streams.py

```diff
@@ -217,9 +220,16 @@ def __exit__(self, exc_type, exc_value, traceback):
         for writer in self._writers:
             writer.__exit__(exc_type, exc_value, traceback)
 
-    def write(self, data):
-        self._writers[self._next_stream].write(data)
-        self._next_stream = (self._next_stream + 1) % len(self._writers)
+    def write(self, data, partition: int | None = None):
```

Comment:
Given how often it's identical we should consider implementing the write method in the base class and overwriting it when needed.